pyogrio/geopandas.py

```diff
@@ -285,11 +292,31 @@ def read_dataframe(
     )
 
     if use_arrow:
+        import pyarrow as pa
+
         meta, table = result
 
         # split_blocks and self_destruct decrease memory usage, but have as side effect
         # that accessing table afterwards causes crash, so del table to avoid.
         kwargs = {"self_destruct": True}
+        if PANDAS_GE_30:
+            # starting with pyarrow 19.0, pyarrow will correctly handle this themselves,
+            # so only use types_mapper as workaround for older versions
+            if not PYARROW_GE_19:
+                kwargs["types_mapper"] = {
+                    pa.string(): pd.StringDtype(na_value=np.nan),
+                    pa.large_string(): pd.StringDtype(na_value=np.nan),
+                    pa.json_(): pd.StringDtype(na_value=np.nan),
+                }.get
+            # TODO enable the below block when upstream issue to accept extension types
```

Comment:
Can this commented block be removed now?

Reply:
I can remove it for now if you prefer, but I plan to do a PR to pandas to fix this there, and then once that is done, I want to add the below code to pyogrio. So I want to uncomment it in a follow-up, but so can also just completely add it in a follow-up ;)

Reply:
No need to remove.  I read it as being covered above but must have been too hasty in my review.